geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/index/QueryPlanner.scala

```diff
@@ -105,63 +106,64 @@ case class QueryPlanner(sft: SimpleFeatureType,
-        } else {
-          // quick check to try to remove duplicate bboxes
-          val filters = (decomposeAnd(densityQuery.getFilter) ++ Seq(bbox)).distinct
-          densityQuery.setFilter(ff.and(filters))
-        }
-        val strategy = QueryStrategyDecider.chooseStrategy(sft, densityQuery, hints, version)
-        val plans = strategy.getQueryPlans(densityQuery, this, output)
-        (plans.map { p => StrategyPlan(strategy, p) }, 1)
-      } else {
-        // As a pre-processing step, we examine the query/filter and split it into multiple queries.
-        // TODO Work to make the queries non-overlapping
-        val splitQueries = splitQueryOnOrs(query, output)
-        val plans = splitQueries.flatMap { q =>
-          val strategy = requested.getOrElse(QueryStrategyDecider.chooseStrategy(sft, q, hints, version))
-          output(s"Strategy: ${strategy.getClass.getCanonicalName}")
-          output(s"Transforms: ${getTransformDefinition(query).getOrElse("None")}")
-          strategy.getQueryPlans(q, this, output).map { plan =>
-            output(s"Table: ${plan.table}")
-            output(s"Column Families${if (plan.columnFamilies.isEmpty) ": all" else s" (${plan.columnFamilies.size}): ${plan.columnFamilies.take(20)}"} ")
-            output(s"Ranges (${plan.ranges.size}): ${plan.ranges.take(5).mkString(", ")}")
-            output(s"Iterators (${plan.iterators.size}): ${plan.iterators.mkString("[", ", ", "]")}")
-            StrategyPlan(strategy, plan)
-          }
-        }
-        (plans, splitQueries.length)
+      val strategies = QueryStrategyDecider.chooseStrategies(sft, query, hints, requested, version)
+      output(s"Strategy count: ${strategies.length}")
+      output(s"Transforms: ${getTransformDefinition(query).getOrElse("None")}")
+      val plans = strategies.flatMap { strategy =>
+        output(s"Strategy: ${strategy.getClass.getCanonicalName}")
+        output(s"Filter: ${strategy.filter.filterString}")
+        val plans = strategy.getQueryPlans(this, query.getHints, output)
+        plans.foreach(outputPlan(_, output))
+        plans
       }
+      (plans, strategies.length)
     }, "plan")
-    output(s"Query planning took ${timings.time("plan")} ms for $numClauses distinct queries.")
+    output(s"Query planning took ${timings.time("plan")}ms for $numClauses distinct queries.")
     (queryPlans, numClauses)
   }
 
+  // output the query plan for explain logging
+  private def outputPlan(plan: QueryPlan, output: ExplainerOutputType, joinPrefix: String = ""): Unit = {
+    output(s"${joinPrefix}Table: ${plan.table}")
+    output(s"${joinPrefix}Column Families${if (plan.columnFamilies.isEmpty) ": all"
+      else s" (${plan.columnFamilies.size}): ${plan.columnFamilies.take(20)}"} ")
+    output(s"${joinPrefix}Ranges (${plan.ranges.size}): ${plan.ranges.take(5).mkString(", ")}")
+    output(s"${joinPrefix}Iterators (${plan.iterators.size}): ${plan.iterators.mkString("[", "],[", "]")}")
```

Comment:
ah, possibly... i think i prefer it this way unless you have a strong preference.